### PR TITLE
Msw update stepped care dates

### PR DESCRIPTION
**Pull request by michaelwehrley**
[#101934888]

### CODE REVIEW

File: app/presenters/think_feel_do_engine/media_access_event_presenter.rb [hunk 1]
```diff
@@ -0,0 +1,57 @@
+module ThinkFeelDoEngine
+  # Simplifies the logic of the media/audio access event views.
+  class MediaAccessEventPresenter
+    attr_reader :event
+
+    def initialize(event)
+      @event = event
+    end
+
+    def completed?
```
Comment (ericcf) [inline app/presenters/think_feel_do_engine/media_access_event_presenter.rb]: Usually methods ending with `?` indicate booleans. Maybe change to `completed_at`?

File: app/presenters/think_feel_do_engine/media_access_event_presenter.rb [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+      end_time
+    end
+
+    def date_when_available
```
Comment (ericcf) [inline app/presenters/think_feel_do_engine/media_access_event_presenter.rb]: For consistency with other model attributes, maybe `available_on`?

File: app/views/think_feel_do_engine/media_access_events/_access_data.html.erb [hunk 3]
```diff
@@ -1,12 +1,24 @@
...
   <tr>
     <td class="not-displayed">
-      <%= (event.participant.active_membership.start_date + event.task_release_day - 1) ? (event.participant.active_membership.start_date + event.task_release_day - 1).to_time.to_i : 0 %>
+      <%= presenter.sortable %>
+    </td>
+    <td><%= presenter.title %></td>
+    <td><%= presenter.date_when_available %></td>
+    <td><%= presenter.formatted_start_time %></td>
+    <td>
+      <% if presenter.completed? %>
```
Comment (ericcf) [inline app/views/think_feel_do_engine/media_access_events/_access_data.html.erb]: Why didn't you put this conditional logic in the presenter itself?

File: spec/dummy/app/models/social_networking/goal.rb [hunk 4]
```diff
@@ -1,5 +1,11 @@
 module SocialNetworking
   class Goal
```
Comment (ericcf) [inline spec/dummy/app/models/social_networking/goal.rb]: Why is this in the project?

File: spec/presenters/think_feel_do_engine/media_access_event_presenter_spec.rb [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+        end_time: Time.zone.now + 1.hour,
+        slide_title: "foo",
+        task_release_day: 2)
+    end
+
+    def media_access_event(event = nil)
+      MediaAccessEventPresenter.new(event)
+    end
+
+    describe "instance methods" do
```
Comment (ericcf) [inline spec/presenters/think_feel_do_engine/media_access_event_presenter_spec.rb]: This outer describe is unnecessary. The `#` conventionally describes instance methods.

File: spec/presenters/think_feel_do_engine/media_access_event_presenter_spec.rb [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+      end
+
+      describe "#formatted_end_time" do
+        it "returns a formatted ending time" do
+          expect(media_access_event(event).formatted_end_time)
+            .to eq event.end_time.to_s(:standard)
+        end
+      end
+
+      describe "#formatted_start_time" do
```
Comment (ericcf) [inline spec/presenters/think_feel_do_engine/media_access_event_presenter_spec.rb]: Do you feel that there is value in writing a spec whose code is identical to the method being tested?

File: spec/views/think_feel_do_engine/coach/patient_dashboards/_inactive_patient.html.erb_spec.rb [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+      context "and patient is stepped" do
+        before do
+          allow(membership).to receive(:stepped_on) { Date.today }
+
+          render partial: partial, locals: { patient: patient }
+        end
+
+        it "displays stepped date" do
+          expect(rendered)
+            .to have_text membership.stepped_on.to_s(:user_date)
```
Comment (ericcf) [inline spec/views/think_feel_do_engine/coach/patient_dashboards/_inactive_patient.html.erb_spec.rb]: I worry that just expecting a date on the page might lead to a false positive. What do you think about testing for it inside of a tag or css class?

**Comment by michaelwehrley** [comment]
@ericcf thanks for the comments!

File: app/presenters/think_feel_do_engine/media_access_event_presenter.rb (hunk 1 above)
Comment (michaelwehrley) [inline app/presenters/think_feel_do_engine/media_access_event_presenter.rb]: Yep.  I add it as `!!end_time` and darn rubocop had me change it.

File: app/presenters/think_feel_do_engine/media_access_event_presenter.rb (hunk 2 above)
Comment (michaelwehrley) [inline app/presenters/think_feel_do_engine/media_access_event_presenter.rb]: I love!

File: app/views/think_feel_do_engine/media_access_events/_access_data.html.erb (hunk 3 above)
Comment (michaelwehrley) [inline app/views/think_feel_do_engine/media_access_events/_access_data.html.erb]: Good catch, my oversight.  I thought I knew more than I did.

File: spec/presenters/think_feel_do_engine/media_access_event_presenter_spec.rb (hunk 6 above)
Comment (michaelwehrley) [inline spec/presenters/think_feel_do_engine/media_access_event_presenter_spec.rb]: Well I am referencing a private method.

File: spec/views/think_feel_do_engine/coach/patient_dashboards/_inactive_patient.html.erb_spec.rb (hunk 7 above)
Comment (michaelwehrley) [inline spec/views/think_feel_do_engine/coach/patient_dashboards/_inactive_patient.html.erb_spec.rb]: :+1:

File: spec/presenters/think_feel_do_engine/media_access_event_presenter_spec.rb (hunk 6 above)
Comment (ericcf) [inline spec/presenters/think_feel_do_engine/media_access_event_presenter_spec.rb]: Where?

Comment (michaelwehrley) [inline spec/presenters/think_feel_do_engine/media_access_event_presenter_spec.rb]: https://github.com/cbitstech/think_feel_do_engine/blob/msw_update_stepped_care_dates/app/presenters/think_feel_do_engine/media_access_event_presenter.rb#L41

Comment (ericcf) [inline spec/presenters/think_feel_do_engine/media_access_event_presenter_spec.rb]: So you do feel that there is value in this unit test?

Comment (michaelwehrley) [inline spec/presenters/think_feel_do_engine/media_access_event_presenter_spec.rb]: Yes, in the sense that the story deals with the change in formatting of dates and times.  However, I agree it isn't the most complicated spec.  Theoretically or practically speaking, do you think it would cause more problems than solutions in the future?

Comment (ericcf) [inline spec/presenters/think_feel_do_engine/media_access_event_presenter_spec.rb]: To me, the value in a unit test comes when there is a complicated piece of code with edge cases or when there are multiple ways of writing the code that may change. In this particular case, if the needs of the code changed, you'd have to change the test to match the code again. What value is being added by duplicating the exact code?

Comment (michaelwehrley) [inline spec/presenters/think_feel_do_engine/media_access_event_presenter_spec.rb]: Sounds reasonable to me.  The test to is too tightly coupled to the code. :white_check_mark:

File: spec/views/think_feel_do_engine/coach/patient_dashboards/_inactive_patient.html.erb_spec.rb (hunk 7 above)
Comment (michaelwehrley) [inline spec/views/think_feel_do_engine/coach/patient_dashboards/_inactive_patient.html.erb_spec.rb]: I absolutely hate adding extra css for testing, so I changed the test to this absurd date: https://github.com/cbitstech/think_feel_do_engine/blob/075622da9bb4d9e51695b4e40e9314f35b96f579/spec/views/think_feel_do_engine/coach/patient_dashboards/_inactive_patient.html.erb_spec.rb#L100

File: spec/dummy/app/models/social_networking/goal.rb (hunk 4 above)
Comment (michaelwehrley) [inline spec/dummy/app/models/social_networking/goal.rb]: It was added for dashboard testing: https://github.com/cbitstech/think_feel_do_engine/commit/6e6cd76ba71e1904879c870e4812e6c067489f57#diff-7971d93852d62871533dbff639549e69. I could remove it and put these tests in social_networking.

File: app/views/think_feel_do_engine/media_access_events/_access_data.html.erb [hunk 8]
```diff
@@ -1,12 +1,24 @@
...
+      <%= presenter.sortable %>
+    </td>
+    <td><%= presenter.title %></td>
+    <td><%= presenter.available_on %></td>
+    <td><%= presenter.formatted_start_time %></td>
+    <td>
+      <% if presenter.completed %>
+        <%= presenter.formatted_end_time %>
+      <% else %>
+        <%= content_tag(:span, "Not Completed", class: ["label", "label-warning"]) %>
```
Comment (ericcf) [inline app/views/think_feel_do_engine/media_access_events/_access_data.html.erb]: Next time, why not just use HTML?
